### DISCUSSION TITLE
Using sendmail-whois-ipmatches but no mail on ban

### DISCUSSION THREAD

**Discussion by BertrandB21** [fail2ban/fail2ban#3902] [Q&A] [answered]
Hello, on debian Trixie
i'm trying to get a complete history for my redicive filter so i have in my jail.local for filter recidive
```
action = nftables-allports[name=recidive]
             sendmail-whois-ipmatches[name=recidive, logpath=/var/log/fail2ban.log]
```
Who is sending me the message but not for old log ratated so i try to change to sendmail-whois-ipmatches but don't get mail on ban.
I do'nt understand why and do'nt see how to debug without poluate the logfile. I tried and crached my vps.

Have you an idea ? ipv6 problem ? bad parameters ?

**Comment by sebres** [comment]
> bad parameters ?

Maybe...
You have to set all required parameters of the action itself, e. g. like jail.conf does it for some prepared interpolations:
https://github.com/fail2ban/fail2ban/blob/a6ca6e2a26b579c5d65049070ac81c298719bba9/config/jail.conf#L215-L216
However without to investigate deeper and without more info (e. g. analyze the logs of fail2ban, etc), it is hard to say what happens here.

Also note that such actions are not recommended at all, e. g. smtp and especially whois may hang up to some timeout, and so may retard the banning drastically (few µs vs. seconds or even minutes), and this all per IP, so a banning may be drastically retarded... 
But still worse it may be affected by a snowball issue: the longer it takes (to be banned) - the more tickets to ban arrive - the more load you get on the system - the more bans occurred - and vicious circle.
Let alone the recidive jail may be additionally worse here - because by some circumstances (of that vicious circle) it may cause a "recursion".
Especially don't do heavy debugging together with recidive jail etc.

Basically a recidive jail is a rudiment (in the way how it is implemented) - better would be to use `bantime.increment = true` instead.

**Reply by BertrandB21** [reply]
I'll try,
I use both bantime.increment and recidive (for very long find delay and very very long bantime). I have only 96 banedip by recidive so no snowball even if it's winter.
How may i trace only the faulty action how to dump the value of ipmatches tag ? if possible of course, create an test_tags action ?

**Reply by BertrandB21** [reply]
How i made an copy paste error in first post it' a problem with sendmail-whois-ipmatches sendmail-whois-lines work fine

**Reply by BertrandB21** [reply, marked as answer]
And now it's work i've just change indentation of action and made an update of debian (i'm in testing) i dunno if it was the problem.
